### ISSUE TITLE
Regression: Artistic License 2.0 (Markdown) no longer detected

### ISSUE THREAD

**Issue by chocolateboy** [licensee/licensee#435]
### Describe the bug

As of [v9.11.0](https://github.com/licensee/licensee/releases/tag/v9.11.0) (Jan 2019), Licensee no longer detects [Artistic License 2.0](https://opensource.org/licenses/artistic-license-2.0) files (Markdown). They were detected correctly prior to [this commit](https://github.com/licensee/licensee/commit/d01660a751977271829339a3d57a181cbac76158).

### Steps to reproduce the behavior

1. Download an Artistic License 2.0 file in Markdown format, [e.g.](https://github.com/chocolateboy/wireless/blob/master/LICENSE.md)
2. Run the test script (see below)

#### Expected

```
ok: Artistic License 2.0
```

#### Actual

```
error: "Other"
```

## test.rb

```ruby
#!/usr/bin/ruby -W0

require 'licensee'

LICENSE = ARGV.shift || 'LICENSE.md'
WANT = 'Artistic License 2.0'

license = Licensee.license(LICENSE)&.name

if license == WANT
  puts "ok: #{license}"
else
  puts "error: #{license.inspect}"
end
```

**Comment by chocolateboy** [comment]
Old output of `licensee` (as of ac07ee788a6aa93a2b29f7f05c685b78dc50884a):

```
License:        Artistic-2.0
Matched files:  LICENSE.md
LICENSE.md:
  Content hash:  68c6b75f4c565c0d2e83203b8a9682cba3756ff4
  Confidence:    99.73%
  Matcher:       Licensee::Matchers::Dice
  License:       Artistic-2.0
  Closest non-matching licenses:
    Artistic-2.0 similarity:  99.73%
```

Current output (1c5595ad09b9178fbb1dc1e3f941d1403dbf8f60):

```
License:        NOASSERTION
Matched files:  LICENSE.md
LICENSE.md:
  Content hash:  7a62390b01fb6e67e6912c453b9095d5ac7eebcc
  License:       NOASSERTION
  Closest non-matching licenses:
    Artistic-2.0 similarity:  95.68%
    Apache-2.0 similarity:    46.23%
    ECL-2.0 similarity:       42.48%
```

**Comment by mlinksva** [comment]
Thanks for reporting this and finding the last good commit for this behavior. Re-running with diff option to see how the normalized text changed.

Old output of licensee (as of ac07ee7):

```
$ cat ../wireless/LICENSE.md |bundle exec licensee diff --license=artistic-2.0
Comparing to Artistic License 2.0:
Input Length:      8570
License length:    8498
Similarity:      99.73%
diff --git a/LICENSE b/LICENSE
index 14b6e1b..4f4dbae 100644
--- a/LICENSE
+++ b/LICENSE
@@ -1,48 +1,48 @@
{+copyright © 2000-2006, the perl foundation.+} everyone is permitted to copy and
distribute verbatim copies of this license document, but changing it is not
allowed. preamble this license establishes the terms under which a given free
software package may be copied, modified, distributed, and/or redistributed. the
intent is that the copyright holder maintains some artistic control over the
development of that package while still keeping the package available as open
source and free software. you are always permitted to make arrangements wholly
outside of this license directly with the copyright holder of a given package.
if the terms of this license do not permit the full use that you propose to make
of the package, you should contact the copyright holder and seek a different
licensing arrangement. definitions "copyright holder" means the individual s or
organization s named in the copyright notice for the entire package.
"contributor" means any party that has contributed code or other material to the
package, in accordance with the copyright holder's procedures. "you" and "your"
means any person who would like to copy, distribute, or modify the package.
"package" means the collection of files distributed by the copyright holder, and
derivatives of that collection and/or of those files. a given package may
consist of either the standard version, or a modified version. "distribute"
means providing a copy of the package or making it accessible to anyone else, or
in the case of a company or organization, to others outside of your company or
organization. "distributor fee" means any fee that you charge for distributing
this package or providing support for this package to another party. it does not
mean licensing fees. "standard version" refers to the package if it has not been
modified, or has been modified only in ways explicitly requested by the
copyright holder. "modified version" means the package, if it has been changed,
and such changes were not explicitly requested by the copyright holder.
"original license" means this artistic license as distributed with the standard
version of the package, in its current version or as it may be modified by the
perl foundation in the future. "source" form means the source code,
documentation source, and configuration files for the package. "compiled" form
means the compiled bytecode, object code, binary, or any other form resulting
from mechanical transformation or translation of the source form. permission for
use and modification without distribution {+-+} 1 you are permitted to use the
standard version and create and use modified versions for any purpose without
restriction, provided that you do not distribute the modified version.
permissions for redistribution of the standard version {+-+} 2 you may distribute
verbatim copies of the source form of the standard version of this package in
any medium without restriction, either gratis or for a distributor fee, provided
that you duplicate all of the original copyright notices and associated
disclaimers. at your discretion, such verbatim copies may or may not include a
compiled form of the package. {+-+} 3 you may apply any bug fixes, portability
changes, and other modifications made available from the copyright holder. the
resulting package will still be considered the standard version, and as such
will be subject to the original license. distribution of modified versions of
the package as source {+-+} 4 you may distribute your modified version as source
either gratis or for a distributor fee, and with or without a compiled form of
the modified version provided that you clearly document how it differs from the
standard version, including, but not limited to, documenting any non-standard
@@ -60,7 +60,7 @@ the same licensing terms that apply to the copy that the licensee received, and
requires that the source form of the modified version, and of any works derived
from it, be made freely available in that license fees are prohibited but
distributor fees are allowed. distribution of compiled forms of the standard
version or modified versions without the source {+-+} 5 you may distribute compiled
forms of the standard version without the source, provided that you include
complete instructions on how to get the source of the standard version. such
instructions must be valid at the time of your distribution. if these
@@ -68,44 +68,44 @@ instructions, at any time while you are carrying out such distribution, become
invalid, you must provide new instructions on demand or cease further
distribution. if you provide valid instructions or cease distribution within
thirty days after you become aware that the instructions are invalid, then you
do not forfeit any of your rights under this license. {+-+} 6 you may distribute a
modified version in compiled form without the source, provided that you comply
with section 4 with respect to the source of the modified version. aggregating
or linking the package {+-+} 7 you may aggregate the package either the standard
version or modified version with other packages and distribute the resulting
aggregation provided that you do not charge a licensing fee for the package.
distributor fees are permitted, and licensing fees for other components in the
aggregation are permitted. the terms of this license apply to the use and
distribution of the standard or modified versions as included in the
aggregation. {+-+} 8 you are permitted to link modified and standard versions with
other works, to embed the package in a larger work of your own, or to build
stand-alone binary or bytecode versions of applications that include the
package, and distribute the result without restriction, provided the result does
not expose a direct interface to the package. items that are not considered part
of a modified version {+-+} 9 works including, but not limited to, modules and
scripts that merely extend or make use of the package, do not, by themselves,
cause the package to be a modified version. in addition, such works are not
considered parts of the package itself, and are not subject to the terms of this
license. general provisions {+-+} 10 any use, modification, and distribution of the
standard or modified versions is governed by this artistic license. by using,
modifying or distributing the package, you accept this license. do not use,
modify, or distribute the package, if you do not accept this license. {+-+} 11 if
your modified version has been derived from a modified version made by someone
other than you, you are nevertheless required to ensure that your modified
version complies with the requirements of this license. {+-+} 12 this license does
not grant you the right to use any trademark, service mark, tradename, or logo
of the copyright holder. {+-+} 13 this license includes the non-exclusive,
worldwide, free-of-charge patent license to make, have made, use, offer to sell,
sell, import and otherwise transfer the package with respect to any patent
claims licensable by the copyright holder that are necessarily infringed by the
package. if you institute patent litigation including a cross-claim or
counterclaim against any party alleging that the package constitutes direct or
contributory patent infringement, then this artistic license to you shall
terminate on the date that such litigation is filed. {+-+} 14 disclaimer of
warranty: the package is provided by the copyright holder and contributors "as
is' and without any express or implied warranties. the implied warranties of
merchantability, fitness for a particular purpose, or non-infringement are
disclaimed to the extent permitted by your local law. unless required by law, no
copyright holder or contributor will be liable for any direct, indirect,
incidental, or consequential damages arising in any way out of the use of the
package, even if advised of the possibility of such damage.
```

Current output (1c5595a):

```
$ bundle exec licensee diff ../wireless/ --license=artistic-2.0
Comparing to Artistic License 2.0:
Input Length:      8640
License length:    8502
Similarity:      95.68%
diff --git a/LICENSE b/LICENSE
index 2032819..8b4776a 100644
--- a/LICENSE
+++ b/LICENSE
@@ -1,111 +1,112 @@
{+copyright © 2000-2006, the perl foundation.+} everyone is permitted to copy and
distribute verbatim copies of this license document, but changing it is not
allowed. {+###+} preamble this license establishes the terms under which a given
free software package may be copied, modified, distributed, and/or
redistributed. the intent is that the copyright holder maintains some artistic
control over the development of that package while still keeping the package
available as open source and free software. you are always permitted to make
arrangements wholly outside of this license directly with the copyright holder
of a given package. if the terms of this license do not permit the full use that
you propose to make of the package, you should contact the copyright holder and
seek a different licensing arrangement. {+###+} definitions "copyright holder" means
the individual(s) or organization(s) named in the copyright notice for the
entire package. "contributor" means any party that has contributed code or other
material to the package, in accordance with the copyright holder's procedures.
"you" and "your" means any person who would like to copy, distribute, or modify
the package. "package" means the collection of files distributed by the
copyright holder, and derivatives of that collection and/or of those files. a
given package may consist of either the standard version, or a modified version.
"distribute" means providing a copy of the package or making it accessible to
anyone else, or in the case of a company or organization, to others outside of
your company or organization. "distributor fee" means any fee that you charge
for distributing this package or providing support for this package to another
party. it does not mean licensing fees. "standard version" refers to the package
if it has not been modified, or has been modified only in ways explicitly
requested by the copyright holder. "modified version" means the package, if it
has been changed, and such changes were not explicitly requested by the
copyright holder. "original license" means this artistic license as distributed
with the standard version of the package, in its current version or as it may be
modified by the perl foundation in the future. "source" form means the source
code, documentation source, and configuration files for the package. "compiled"
form means the compiled bytecode, object code, binary, or any other form
resulting from mechanical transformation or translation of the source form. {+###+}
permission for use and modification without distribution [-*-]{+(1)+} you are permitted
to use the standard version and create and use modified versions for any purpose
without restriction, provided that you do not distribute the modified version.
{+###+} permissions for redistribution of the standard version [-*-]{+(2)+} you may
distribute verbatim copies of the source form of the standard version of this
package in any medium without restriction, either gratis or for a distributor
fee, provided that you duplicate all of the original copyright notices and
associated disclaimers. at your discretion, such verbatim copies may or may not
include a compiled form of the package. [-*-]{+(3)+} you may apply any bug fixes,
portability changes, and other modifications made available from the copyright
holder. the resulting package will still be considered the standard version, and
as such will be subject to the original license. {+###+} distribution of modified
versions of the package as source [-*-]{+(4)+} you may distribute your modified version
as source (either gratis or for a distributor fee, and with or without a
compiled form of the modified version) provided that you clearly document how it
differs from the standard version, including, but not limited to, documenting
any non-standard features, executables, or modules, and provided that you do at
least one of the following: [-*-]{+- (a)+} make the modified version available to the
copyright holder of the standard version, under the original license, so that
the copyright holder may include your modifications in the standard version. [-*-]{+-+}
{+(b)+} ensure that installation of your modified version does not prevent the user
installing or running the standard version. in addition, the modified version
must bear a name that is different from the name of the standard version. [-*-]{+- (c)+}
allow anyone who receives a copy of the modified version to make the source form
of the modified version available to others under [-*-]{+- (i)+} the original license or
[-*-]{+- (ii)+} a license that permits the licensee to freely copy, modify and
redistribute the modified version using the same licensing terms that apply to
the copy that the licensee received, and requires that the source form of the
modified version, and of any works derived from it, be made freely available in
that license fees are prohibited but distributor fees are allowed. {+###+}
distribution of compiled forms of the standard version {+###+} or modified versions
without the source [-*-]{+(5)+} you may distribute compiled forms of the standard version
without the source, provided that you include complete instructions on how to
get the source of the standard version. such instructions must be valid at the
time of your distribution. if these instructions, at any time while you are
carrying out such distribution, become invalid, you must provide new
instructions on demand or cease further distribution. if you provide valid
instructions or cease distribution within thirty days after you become aware
that the instructions are invalid, then you do not forfeit any of your rights
under this license. [-*-]{+(6)+} you may distribute a modified version in compiled form
without the source, provided that you comply with section 4 with respect to the
source of the modified version. {+###+} aggregating or linking the package [-*-]{+(7)+} you
may aggregate the package (either the standard version or modified version) with
other packages and distribute the resulting aggregation provided that you do not
charge a licensing fee for the package. distributor fees are permitted, and
licensing fees for other components in the aggregation are permitted. the terms
of this license apply to the use and distribution of the standard or modified
versions as included in the aggregation. [-*-]{+(8)+} you are permitted to link modified
and standard versions with other works, to embed the package in a larger work of
your own, or to build stand-alone binary or bytecode versions of applications
that include the package, and distribute the result without restriction,
provided the result does not expose a direct interface to the package. {+###+} items
that are not considered part of a modified version [-*-]{+(9)+} works (including, but not
limited to, modules and scripts) that merely extend or make use of the package,
do not, by themselves, cause the package to be a modified version. in addition,
such works are not considered parts of the package itself, and are not subject
to the terms of this license. {+###+} general provisions [-*-]{+(10)+} any use, modification,
and distribution of the standard or modified versions is governed by this
artistic license. by using, modifying or distributing the package, you accept
this license. do not use, modify, or distribute the package, if you do not
accept this license. [-*-]{+(11)+} if your modified version has been derived from a
modified version made by someone other than you, you are nevertheless required
to ensure that your modified version complies with the requirements of this
license. [-*-]{+(12)+} this license does not grant you the right to use any trademark,
service mark, tradename, or logo of the copyright holder. [-*-]{+(13)+} this license
includes the non-exclusive, worldwide, free-of-charge patent license to make,
have made, use, offer to sell, sell, import and otherwise transfer the package
with respect to any patent claims licensable by the copyright holder that are
necessarily infringed by the package. if you institute patent litigation
(including a cross-claim or counterclaim) against any party alleging that the
package constitutes direct or contributory patent infringement, then this
artistic license to you shall terminate on the date that such litigation is
filed. [-*-]{+(14)+} disclaimer of warranty: the package is provided by the copyright
holder and contributors "as is' and without any express or implied warranties.
the implied warranties of merchantability, fitness for a particular purpose, or
non-infringement are disclaimed to the extent permitted by your local law.
unless required by law, no copyright holder or contributor will be liable for
any direct, indirect, incidental, or consequential damages arising in any way
out of the use of the package, even if advised of the possibility of such
damage.
```

It looks like current licensee does a worse job of normalizing away both markdown headings and markdown bold. Don't have an immediate idea on the former, but on the latter, I guess there might be some sensitivity to order of normalizing bullets and bolding. The first non-working commit you pointed out https://github.com/licensee/licensee/commit/d01660a751977271829339a3d57a181cbac76158 is a refactoring of content normalization, so it isn't a huge surprise that it also introduced a regression.

The file you provided https://github.com/chocolateboy/wireless/blob/master/LICENSE.md looks like a good one to add as a test fixture for licensee.

All versions fail to normalize out the italicized Perl Foundation copyright notice. That likely should be fixed as part of fixing #430.

**Comment by chocolateboy** [comment]
This is still an issue.
